Signal/src/ViewControllers/FingerprintViewController.m

```diff
@@ -63,59 +80,234 @@ - (void)configureWithRecipientId:(NSString *)recipientId
         [builder fingerprintWithTheirSignalId:recipientId theirIdentityKey:recipientIdentity.identityKey];
 }
 
-- (void)viewDidLoad
+- (void)loadView
 {
-    [super viewDidLoad];
+    [super loadView];
+
+    self.title = NSLocalizedString(@"PRIVACY_VERIFICATION_TITLE", @"Navbar title");
 
-    self.navigationItem.leftBarButtonItem = self.dismissModalButton;
+    self.navigationItem.leftBarButtonItem =
+        [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemStop
+                                                      target:self
+                                                      action:@selector(closeButton)];
+    self.shareButton = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemAction
+                                                                     target:self
+                                                                     action:@selector(didTapShareButton)];
     self.navigationItem.rightBarButtonItem = self.shareButton;
-    [self.modalNavigationBar pushNavigationItem:self.navigationItem animated:NO];
-
-    // HACK for transparent navigation bar.
-    [self.modalNavigationBar setBackgroundImage:[UIImage new] forBarMetrics:UIBarMetricsDefault];
-    self.modalNavigationBar.shadowImage = [UIImage new];
-    self.modalNavigationBar.translucent = YES;
-
-    // HACK to get full width preview layer
-    CGRect oldFrame = self.qrScanningView.frame;
-    CGRect newFrame = CGRectMake(oldFrame.origin.x,
-        oldFrame.origin.y,
-        self.view.frame.size.width,
-        self.view.frame.size.height / 2.0f - oldFrame.origin.y);
-    self.qrScanningView.frame = newFrame;
-    // END HACK to get full width preview layer
 
-    self.title = NSLocalizedString(@"PRIVACY_VERIFICATION_TITLE", @"Navbar title");
-    NSString *instructionsFormat = NSLocalizedString(@"PRIVACY_VERIFICATION_INSTRUCTIONS",
-        @"Paragraph(s) shown alongside the safety number when verifying privacy with {{contact name}}");
-    self.instructionsLabel.text = [NSString stringWithFormat:instructionsFormat, self.contactName];
+    [self createViews];
+}
 
-    NSString *scanTitle = NSLocalizedString(@"SCAN_CODE_ACTION",
-        @"Button label presented with camera icon while verifying privacy credentials. Shows the camera interface.");
-    [self.scanButton setTitle:scanTitle forState:UIControlStateNormal];
-    self.scanningInstructions.text
+- (void)createViews
+{
+    UIColor *darkGrey = [UIColor colorWithRGBHex:0x404040];
```

Comment:
I'm currently using dark grey for the text and circle.  This will probably change.